cfme/infrastructure/virtual_machines.py

```diff
@@ -86,46 +88,174 @@
 retire_remove_button = "//span[@id='remove_button']/a/img"
 
 match_page = partial(match_location, controller='vm_infra', title='Virtual Machines')
-vm_templates_tree = partial(accordion.tree, "VMs & Templates")
-vms_tree = partial(accordion.tree, "VMs")
-templates_tree = partial(accordion.tree, "Templates")
-
-
-def reset_page():
-    tb.select("Grid View")
-    from cfme.web_ui import paginator
-    if sel.is_displayed(search_box.search_field):
-        search.ensure_normal_search_empty()
-    if paginator.page_controls_exist():
-        # paginator.results_per_page(1000)
-        paginator.check_all()
-        paginator.uncheck_all()
 
 
 drift_table = CheckboxTable("//th[normalize-space(.)='Timestamp']/ancestor::table[1]")
 
 
-class InfraVmDetailsToolbar(View):
-    configuration = Dropdown("Configuration")
-    policy = Dropdown("Policy")
-    lifecycle = Dropdown("Lifecycle")
+class InfraVmTemplatesGenericDetailsToolbar(View):
```

Comment:
**Optional**
Consider making these class names a bit shorter, I think `VmTemplatesDetailsToolbar` is fine here, and just `Vm` in place of `InfraVm` in the class names below.

At least all consistent capitalization - there's an `InfraVM` on L132

Reply:
On second thought, it looks like there are only slight differences between this and VMToolbar, I believe a chunk of this might get updated again with the common.vm refactor.

Reply:
I had to make Generic toolbar for templates and vms. when we'll be merging it with instances it can be refactored